src/components/notification/react/NotificationProvider.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState } from 'react';
+
+import { IGenericProps } from 'LumX/core/react/utils';
+
+import { INITIAL_STATE } from 'LumX/components/notification/constants';
+import { NotificationProps, NotificationTypes } from 'LumX/components/notification/react/Notification';
+import { NotificationHandler } from 'LumX/components/notification/react/NotificationHandler';
+
+/////////////////////////////
+
+/**
+ * State of notification.
+ */
+interface INotificationState {
+    props: NotificationProps;
+    close(): void;
+    open(props: NotificationProps): void;
+    error(props: NotificationProps): void;
+    info(props: NotificationProps): void;
+    success(props: NotificationProps): void;
+    warning(props: NotificationProps): void;
+}
+type NotificationState = INotificationState;
+
+/**
+ * React context of notification.
+ */
+const notificationContext: React.Context<INotificationState> = React.createContext(INITIAL_STATE);
+
+/////////////////////////////
+
+/**
+ * Defines the props of the component.
+ */
+interface INotificationProviderProps extends IGenericProps {}
+type NotificationProviderProps = INotificationProviderProps;
+
+/////////////////////////////
+
+/**
+ * Notification context provider.
+ *
+ * @return {React.ReactElement} The component.
+ */
+const NotificationProvider: React.FC<NotificationProviderProps> = ({
+    children,
+}: NotificationProviderProps): React.ReactElement => {
+    const [notificationState, setNotificationState]: [
+        NotificationState,
+        React.Dispatch<React.SetStateAction<NotificationState>>
+    ] = useState<NotificationState>({
```

Comment:
I think you could have used `useReducer` + an `immerjs` approach to get the code cleaner.

Reply:
Yep you're right, i should have. But at first, i didn't want to implement a state management like.
I will take a look at this